ports/psoc6/main.c

```diff
@@ -36,70 +36,53 @@
 // port-specific includes
 #include "mplogger.h"
 
-/* Task header files */
-#include "scan_task.h"
+#define MPY_TASK_STACK_SIZE                    (4096u)
+#define MPY_TASK_PRIORITY                      (3u)
 
+void mpy_task(void *arg);
 
-/* Include serial flash library and QSPI memory configurations only for the
- * kits that require the Wi-Fi firmware to be loaded in external QSPI NOR flash.
- */
-#if defined(CY_DEVICE_PSOC6A512K)
-#include "cy_serial_flash_qspi.h"
-#include "cycfg_qspi_memslot.h"
-#endif
-
+TaskHandle_t mpy_task_handle;
 
-// /*******************************************************************************
-//  * Macros
-//  ******************************************************************************/
-#define GPIO_INTERRUPT_PRIORITY             (7)
+int main(int argc, char **argv) {
+    // Initialize the device and board peripherals
+    cy_rslt_t result = cybsp_init();
+    if (result != CY_RSLT_SUCCESS) {
+        mp_raise_ValueError(MP_ERROR_TEXT("cybsp_init failed !\n"));
+    }
 
+    // Initialize retarget-io to use the debug UART port
+    result = cy_retarget_io_init(CYBSP_DEBUG_UART_TX, CYBSP_DEBUG_UART_RX, CY_RETARGET_IO_BAUDRATE);
+    if (result != CY_RSLT_SUCCESS) {
+        mp_raise_ValueError(MP_ERROR_TEXT("cy_retarget_io_init failed !\n"));
+    }
 
-// /*******************************************************************************
-//  * Global Variables
-//  ******************************************************************************/
-cyhal_gpio_callback_data_t cb_data =
-{
-    .callback = gpio_interrupt_handler,
-    .callback_arg = NULL
-};
+    xTaskCreate(mpy_task, "MicroPython task", MPY_TASK_STACK_SIZE, NULL, MPY_TASK_PRIORITY, &mpy_task_handle);
+    vTaskStartScheduler();
 
+    // Should never get here
+    CY_ASSERT(0);
+    return 0;
+}
 
 #if MICROPY_ENABLE_GC
```

Comment:
I would move the following defs and extern statements to the top before main to stick to common practice.

Reply:
done